spark3/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -160,6 +167,43 @@ public WriteBuilder newWriteBuilder(LogicalWriteInfo info) {
     return new SparkWriteBuilder(sparkSession(), icebergTable, info);
   }
 
+  @Override
+  public MergeBuilder newMergeBuilder(LogicalWriteInfo info) {
```

Comment:
We could set `operation` inside `info.options()` in `RewriteDelete` and pass it here so that we can support table properties like `write.delete.mode` instead of `write.row-level.mode`. Thoughts, @rdblue?

Reply:
I'm not a fan of passing things through Spark's options. If we have the information in the writer, can we just call a setter to set it on the reader?

Reply:
We have this info only in `RewriteDelete` rule that calls `newMergeBuilder`. After that, we have no way to know whether it is a delete or merge.

Reply:
So if we pass something that tells us what logical operation it is, we can have `write.delete.isolation-level`.

Reply:
Unless we want to have `newMergeBuilder`, `newDeleteBuilder`, etc.

Reply:
I like that this is used for both MERGE and DELETE without needing to customize, so I'm reluctant to do too much here. It is close, but I think I would opt to have a single config property.

Operation is something that we could pass through `LogicalWriteInfo` in the future for better logging and purposes like this one. For now, we could add the operation as a String passed to `newMergeBuilder`, but there is no guarantee that we would add this to `LogicalWriteInfo` later. I think that means that we should go for a single property and we can customize later if we do get the operation when the builder is created.

Reply:
After discussing this directly, I think the right way to go is to add the operation name. That fixes the isolation level and mode property problem. And it is reasonable to add the operation to `LogicalWriteInfo`.

Reply:
Done.